packages/sit-onyx/src/components/OnyxSlider/OnyxSlider.stories.ts

```diff
@@ -0,0 +1,316 @@
+ */
+export const CustomError = {
+  args: {
+    ...Default.args,
+    label: "Custom Error",
+    showError: true,
+    customError: {
+      shortMessage: "Custom error",
+      longMessage: "This text might inform the users what they can do to fix the error.",
+    },
+  },
+} satisfies Story;
+
+/**
+ * This example shows a slider with a message / help text at the bottom.
+ */
+export const WithMessage = {
+  args: {
+    ...Default.args,
+    message: { shortMessage: "Example message" },
+  },
+} satisfies Story;
+
+/**
+ * This example shows a skeleton slider.
+ */
+export const Skeleton = {
+  args: {
+    ...Default.args,
+    skeleton: true,
+  },
+} satisfies Story;
+
+/**
+ * This example shows the slider with two thumbs.
+ */
+export const TwoThumbs = {
+  args: {
+    modelValue: [20, 75],
+    label: "Two Thumbs",
+  },
+} satisfies Story;
+
+/**
+ * This example shows the slider with two thumbs.
+ */
+export const ThreeThumbs = {
+  args: {
+    modelValue: [20, 30, 40],
+    label: "Three Thumbs",
+  },
+} satisfies Story;
+
+/**
+ * This example shows the disabled state of the slider.
+ */
+export const Disabled = {
+  args: {
+    disabled: true,
+    modelValue: [20, 40],
+    label: "Disabled",
+  },
+} satisfies Story;
+
+/**
+ * This example shows the slider with marks.
+ */
+export const WithMarks = {
+  args: {
+    modelValue: [20, 40],
+    marks: [0, 25, 50, 75, 100],
+    label: "With Marks",
+  },
+} satisfies Story;
+
+/**
+ * This example shows the slider with marks and labels.
+ */
+export const WithLabelledMarks = {
+  args: {
+    modelValue: [20, 40],
+    marks: [
+      { value: 0, label: "0°C" },
+      { value: 25, label: "25°C" },
+      { value: 50, label: "50°C" },
+      { value: 75, label: "75°C" },
+      { value: 100, label: "100°C" },
+    ],
+    label: "With Labelled Marks",
+    message: { shortMessage: "Example message" },
+  },
+} satisfies Story;
+
+/**
+ * This example shows the disabled slider with marks and labels.
+ */
+export const DisabledWithLabelledMarks = {
+  args: {
+    disabled: true,
+    modelValue: [20, 40],
+    marks: [
+      { value: 0, label: "0°C" },
+      { value: 25, label: "25°C" },
+      { value: 50, label: "50°C" },
+      { value: 75, label: "75°C" },
+      { value: 100, label: "100°C" },
+    ],
+    label: "Disabled With Labelled Marks",
+  },
+} satisfies Story;
+
+/**
+ * This example shows the slider with automatically generated marks.
+ * The marks are generated based on the `step` property.
+ * If `marks` is `true`, marks will be shown for each step.
+ */
+export const AutomaticallyGeneratedMarks = {
+  args: {
+    modelValue: [20, 40],
+    step: 10,
+    marks: true,
+    label: "Automatically Generated Marks",
+  },
+} satisfies Story;
+
+/**
+ * This example shows the discrete state of the slider with marks.
+ * The step is set to null, so the thumbs can only be moved to the marks.
+ */
+export const Discrete = {
```

Comment:
Question: Whats the API currently to have "discrete" steps?
If I see it correctly the users must pass marks and set the step to null, right? Would be great if we can find an API that also allows auto-generating the marks (which requires the step property).

Maybe we can make this more explicit and add a new property `discrete?: boolean`